### ISSUE TITLE
_nftContract can be arbitrarily set

### ISSUE THREAD

**Issue by vv1nn** [code-423n4/2023-03-canto-identity#3]
`    function mint(address _nftContract, uint256 _nftID) external {
        uint256 tokenId = ++numMinted;
        if (ERC721(_nftContract).ownerOf(_nftID) != msg.sender)
            revert PFPNotOwnedByCaller(msg.sender, _nftContract, _nftID);
        ProfilePictureData storage pictureData = pfp[tokenId];
        pictureData.nftContract = _nftContract;
        pictureData.nftID = _nftID;
        _mint(msg.sender, tokenId);
        emit PfpAdded(msg.sender, tokenId, _nftContract, _nftID);
    }`

The mint function in the above code seems to not be limited to a particular NFT collection or contract. It can be used with any NFT contract that is ERC721 compliant, as the _nftContract argument accepts any address.

So long a user has ownership of a specific NFT within an ERC721 compliance, they can use it to mint a new profile picture token with this mint function.

In summary, the mint function can be used with any ERC721-compliant contract and is not limited to a specific collection or contract.

**Comment by vv1nn** [comment]
-
